libsolidity/interface/CompilerStack.cpp

```diff
@@ -547,8 +561,6 @@ void CompilerStack::link()
 
 vector<string> CompilerStack::contractNames() const
 {
-	if (m_stackState < AnalysisPerformed)
-		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Parsing was not successful."));
 	vector<string> contractNames;
 	for (auto const& contract: m_contracts)
```

Comment:
Is this map really initialized early enough?

Reply:
No, it returns an empty vector here. but that doesn't seem to impact the generation of an AST

Reply:
Can you still fix it, please?

Reply:
Either you have to keep the assertion about the state or create the map.

Reply:
It's not easy to create the map as the AST is only populated with the required information in the analysis() step (the `registerDeclaration()` step).